vm.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+package procfs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+// The VM interface is described at
+//   https://www.kernel.org/doc/Documentation/sysctl/vm.txt
+// Each setting is exposed as a single file.
+// Each file contains one line with a single numerical value, except lowmem_reserve_ratio which holds an array
+// and numa_zonelist_order (deprecated) which is a string
+type Vm struct {
+	AdminReserveKbytes        int64 `fileName:"admin_reserve_kbytes"`
+	BlockDump                 int64 `fileName:"block_dump"`
+	CompactUnevictableAllowed int64 `fileName:"compact_unevictable_allowed"`
+	DirtyBackgroundBytes      int64 `fileName:"dirty_background_bytes"`
+	DirtyBackgroundRatio      int64 `fileName:"dirty_background_ratio"`
+	DirtyBytes                int64 `fileName:"dirty_bytes"`
+	DirtyExpireCentisecs      int64 `fileName:"dirty_expire_centisecs"`
+	DirtyRatio                int64 `fileName:"dirty_ratio"`
+	DirtytimeExpireSeconds    int64 `fileName:"dirtytime_expire_seconds"`
+	DirtyWritebackCentisecs   int64 `fileName:"dirty_writeback_centisecs"`
+	DropCaches                int64 `fileName:"drop_caches"`
+	ExtfragThreshold          int64 `fileName:"extfrag_threshold"`
+	HugetlbShmGroup           int64 `fileName:"hugetlb_shm_group"`
+	LaptopMode                int64 `fileName:"laptop_mode"`
+	LegacyVaLayout            int64 `fileName:"legacy_va_layout"`
+	// LowmemReserveRatio        [?]int64 `fileName:"lowmem_reserve_ratio"`
```

Comment:
Looks like this can be parsed into a Go slice.

Reply:
as I wrote already I am not quite sure what possible formats this can have. What I have seen so far are 3 or 5 values, but it seems depending on numa zones this could be much larger?

Reply:
Seems reasonable to me to parse whatever's in here as `[]int64`.